endpointmanager/pkg/archivefile/archivefile.go

```diff
@@ -0,0 +1,530 @@
+}
+type httpResponse struct {
+	ResponseCode  int `json:"http_response_code"`
+	ResponseCount int `json:"http_response_count"`
+}
+type smartHTTPResponse struct {
+	ResponseCode  int `json:"smart_http_response_code"`
+	ResponseCount int `json:"smart_http_response_count"`
+}
+type responseErrors struct {
+	Error      string `json:"error"`
+	ErrorCount int    `json:"error_count"`
+}
+
+// Result is the value that is returned from getting the history data from the
+// given URL
+type Result struct {
+	URL     string
+	Summary totalSummary
+}
+
+// historyArgs is the format for the data passed to getHistory from a worker
+type historyArgs struct {
+	fhirURL   string
+	dateStart string
+	dateEnd   string
+	store     *postgresql.Store
+	result    chan Result
+}
+
+// historyEntry is the format of the data received from the history table for the given URL
+type historyEntry struct {
+	URL              string
+	UpdatedAt        time.Time
+	Operation        string
+	FHIRVersion      string
+	FHIRVersionError error
+	TLSVersion       string
+	MIMETypes        []string
+}
+
+// vendorEntry is the format of the data received from the vendor table for the given URL
+type vendorEntry struct {
+	URL        string
+	VendorName string
+}
+
+// metadataEntry is the format of the data received from the fhir_endpoints_metadata for the
+// given URL
+type metadataEntry struct {
+	URL                 string
+	ResponseTimeSeconds float64
+	HTTPResponse        int
+	SMARTHTTPResponse   int
+	Errors              string
+}
+
+// CreateArchive gets all data from fhir_endpoints, fhir_endpoints_info and vendors between
+// the given start and end date and summarizes the data
+func CreateArchive(ctx context.Context, store *postgresql.Store, dateStart string, dateEnd string) ([]totalSummary, error) {
+	// Get the fhir_endpoints specific information
+	sqlQuery := "SELECT DISTINCT url, organization_names, created_at, list_source from fhir_endpoints;"
+	rows, err := store.DB.QueryContext(ctx, sqlQuery)
+	if err != nil {
+		return nil, fmt.Errorf("ERROR getting data from fhir_endpoints: %s", err)
+	}
+
+	var urls []string
+	allData := make(map[string]totalSummary)
+	defer rows.Close()
+	for rows.Next() {
+		var entry totalSummary
+		var listSource string
+		err = rows.Scan(
+			&entry.URL,
+			pq.Array(&entry.OrganizationNames),
+			&entry.CreatedAt,
+			&listSource)
+		if err != nil {
+			return nil, fmt.Errorf("ERROR getting row from fhir_endpoints: %s", err)
+		}
+
+		// If the URL already exists, include the new list source and organization names
+		if val, ok := allData[entry.URL]; ok {
+			val.ListSource = append(val.ListSource, listSource)
+			val.OrganizationNames = append(val.OrganizationNames, entry.OrganizationNames...)
+			allData[entry.URL] = val
+		} else {
+			entry.ListSource = []string{listSource}
+			allData[entry.URL] = entry
+			urls = append(urls, entry.URL)
+		}
+	}
+
+	errs := make(chan error)
+	numWorkers := viper.GetInt("export_numworkers")
```

Comment:
I think we typically pull the config type of variables in the "main.go"s and pass what is relevant to the function. If you do this, you can also avoid the `if numWorkers == 0` check because the ` config.SetupConfig()` call in the main.go will handle setting the defaults

Reply:
yeah you're right, i've moved them out to the main file!